integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/QueueTestCase.java

```diff
@@ -108,9 +129,106 @@ public void performSingleQueueSendReceiveTestCase() throws JMSException, NamingE
         queueReceiver.closeReceiver();
 
         Assert.assertEquals(receivedMessageCount, sendMessageCount, "assertion failed. Expected message count : "
-                            + sendMessageCount + ".Received message count : " + receivedMessageCount);
+                + sendMessageCount + ".Received message count : " + receivedMessageCount);
+    }
+
+    /**
+     * 1. Create 2 consumers for simple queue
+     * 2. Publish 3000 message to queue
+     * 3. Total messages received by both consumers should be 3000 messages.
+     *
+     * @throws JMSException
+     * @throws NamingException
+     * @throws InterruptedException
+     * @throws IOException
+     */
+    @Test
+    public void performManyConsumersTestCase() throws JMSException, NamingException, InterruptedException, IOException {
+        int sendCount = 3000;
+        int expectedCount = 3000;
+
+        //create consumer 1
+        QueueReceiver queueReceiver1 = new QueueReceiver("manyConsumersTestQueue", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueReceiver1.registerSubscriber();
+
+        TimeUnit.SECONDS.sleep(50L);
+
+        //create consumer 2
+        QueueReceiver queueReceiver2 = new QueueReceiver("manyConsumersTestQueue", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueReceiver2.registerSubscriber();
+
+        TimeUnit.SECONDS.sleep(50L);
+
+        //create publisher
+        QueueSender queueSender = new QueueSender("manyConsumersTestQueue", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueSender.sendMessages(sendCount, "text message");
+
+        TimeUnit.SECONDS.sleep(50L);
+
+        int receivedMessageCount = queueReceiver1.receivedMessageCount() + queueReceiver2.receivedMessageCount();
+
+        // close queue sender.
+        queueSender.closeSender();
```

Comment:
added resource cleans in finally blocks in 06072de2be2dab2bc10d61cad43337340f61fe71